fcrepo4/fcrepo4
# Fcrepo 2747: Advertising External Content Support

**JIRA Ticket**: https://jira.duraspace.org/browse/FCREPO-2747

# What does this Pull Request do?
Adds Accept-External-Content-Handling header to OPTIONS response for binary resources and updates unit tests to ensure header is added only for requests to binary resources.

# What's new?
Adds new header inside FedoraLdp.addLinkAndOptionsHttpHeaders()
Updates unit tests in FedoraLdpTest

# How should this be tested?
In addition to running the FedoraLdpTest suite, you can curl an OPTIONS request to a binary resource and inspect the response headers. "Accept-External-Content-Handling: copy,redirect" should be present among those headers. You can also curl an OPTIONS request to a non-binary resource and verify that the same header does not appear.

E.g., for binary: `curl -i -X POST -H "Content-Type:text/plain" "http://localhost:8080/rest"`
                         `curl -i -X OPTIONS http://localhost:8080/rest/<path/to/binary/resource>`

For non-binary: `curl -X POST "http://localhost:8080/rest/"`
                         `curl -i -X OPTIONS http://localhost:8080/rest/<path/to/resource>`

# Interested parties
@dbernstein 
@bseeger
